### ISSUE TITLE
[systemd] no error if port already in use

### ISSUE THREAD

**Issue by cocoon** [neutrinolabs/xrdp#644]
When I try to start a second instance with the same port configured in xrdp.ini for example with:
`systemctl start xrdp2.service`

It shows no error message but will not be running.
You can see the error in:
 ` journalctl -xe`

`[ERROR] xrdp_listen_main_loop: listen error, possible port already in use`

(Ubuntu 16.04)

**Comment by metalefty** [comment]
I confiemd this issue.

**Comment by metalefty** [comment]
This issue is caused xrdp process exits normally even if `xrdp_listen_main_loop` returns with failure.
I'm working on this issue.

**Comment by cocoon** [comment]
Thanks for the great work you are all doing to get this stable on modern systems!

**Comment by metalefty** [comment]
Also, xrdp-sesman service should raise error if 3350 port is already in use.

**Comment by metalefty** [comment]
@cocoon I've submitted a fix. Can you test?

**Comment by cocoon** [comment]
Great, looks good.
The only thing I would add to xrdp is the same hint where to check for the exact error like for xrdp-sesman:

**Currently:**
```
# systemctl start xrdp-sesman
Job for xrdp-sesman.service failed because the control process exited with error code. See "systemctl status xrdp-sesman.service" and "journalctl -xe" for details.

# systemctl start xrdp
Job for xrdp.service canceled.
```

**Improved:**
```
# systemctl start xrdp
Job for xrdp.service canceled. See "systemctl status xrdp.service" and "journalctl -xe" for details.
```

**Comment by metalefty** [comment]
The message is shown by systemd, not by xrdp. It cannot be controlled by xrdp code. Probably, the difference of these messages might be caused by service dependency. Both 3350/tcp and 3389/tcp is in use, the message is not "Job for xrdp.service canceled.".

```
$ nc -l 3350 & nc -l 3389 &
$ sudo systemctl start xrdp
(see what is shown)
```


I'm not sure about systemd services and what I can do to implement what you want.

**Comment by cocoon** [comment]
Oh yes sorry you're right.
I was so sure it must be possible to control that but I found that request for systemd:
https://github.com/systemd/systemd/issues/3448

**Comment by metalefty** [comment]
Fixed by #753.
